utils/gen-snapshot-tests.js

```diff
@@ -95,41 +96,100 @@ function generateTestFile(root, examples, componentFileMap) {
     console.log(`wrote ${outPath}`);
 }
 
-for (const section of styleguideConfig.sections) {
-    const content = fs.readFileSync(section.content, "utf8");
+function readExamplesFromDocument(documentPath) {
+    if (!fs.existsSync(documentPath)) {
+        return [];
+    }
+    const content = fs.readFileSync(documentPath, "utf8");
     const tokens = marked.lexer(content);
     const examples = tokens
         .filter((token) => token.type === "code")
         .map((token) => token.text);
-    if (examples.length === 0) {
-        // eslint-disable-next-line no-console
-        console.warn(`no examples for section ${section.name}`);
-        continue;
+
+    return examples;
+}
+
+function extractExamplesAndComponentsForFile(file, collation) {
+    const src = fs.readFileSync(file, "utf8");
+    const match =
+        src.match(/export default class \s*(\w+)/) ||
+        src.match(/export default \s*\w+\(\s*(\w+)/);
```

Comment:
This is to support things that aren't classes?

Reply:
Yeah, that was already there in the old code. It is grabbing things like our stuff that gets wrapped by MediaLayout (FixedCell, for example).

Reply:
right, I remember now... man my memory is bad 😅